app/controllers/cafe_works_controller.rb

```diff
@@ -0,0 +1,84 @@
+# encoding:UTF-8
+class CafeWorksController < ApplicationController
+  before_action :set_cafe_work, except: [:main, :nyckelpiga]
+  before_action :councils, except: [:main, :nyckelpiga]
+  before_action :nyckelpiga_auth, only: :nyckelpiga
+  before_action :set_faqs, only: :main
+
+  def show
+    @cwork.load(current_user)
+    @path = (@cwork.has_worker?) ? :update_worker : :add_worker
+  end
+
+  def authorize
+    @authenticated = @cwork.authorize(worker_params[:access_code])
+    @path = :update_worker
+  end
+
+  def update_worker
+    if @cwork.update_worker(worker_params, current_user)
+      flash[:notice] = "Bokningen uppdaterades"
+      redirect_to @cwork
+    else
+      render action: :show
+    end
+  end
+
+  def add_worker
+    flash[:notice] = "Jobbare lades till" if @cwork.add_worker(worker_params, current_user)
```

Comment:
Prefer single-quoted strings when you don't need string interpolation or special symbols.